dask/array/fft.py

```diff
@@ -24,21 +24,24 @@
     """
 
 
-def _fft_wrap(fft_func, dtype, out_chunk_fn):
-    def func(a, n=None, axis=-1):
-        if len(a.chunks[axis]) != 1:
-            raise ValueError(chunk_error % (axis, a.chunks[axis]))
+def _fft_wrap(dtype, out_chunk_fn):
+    def _fft_wrapper(fft_func):
+        def func(a, n=None, axis=-1):
+            if len(a.chunks[axis]) != 1:
+                raise ValueError(chunk_error % (axis, a.chunks[axis]))
 
-        chunks = out_chunk_fn(a, n, axis)
+            chunks = out_chunk_fn(a, n, axis)
 
-        return map_blocks(partial(fft_func, n=n, axis=axis), a, dtype=dtype,
-                          chunks=chunks)
+            return map_blocks(partial(fft_func, n=n, axis=axis), a, dtype=dtype,
+                              chunks=chunks)
```

Comment:
It is useful to avoid partial.  map_blocks can accept keyword arguments, I recommend the following:

    map_blocks(fft_func, a, n=n, axis=axis, dtype=dtype, chunks=chunks)